src/metabase/lib/schema/aggregation.cljc

```diff
@@ -1,47 +1,76 @@
 (ns metabase.lib.schema.aggregation
   (:require
+   [clojure.string :as str]
    [metabase.lib.hierarchy :as lib.hierarchy]
    [metabase.lib.schema.expression :as expression]
    [metabase.lib.schema.mbql-clause :as mbql-clause]
    [metabase.shared.util.i18n :as i18n]
    [metabase.util.malli.registry :as mr]))
 
+;;; `:metric` refs are allowed as aggregations. The definition for `:metric` is in [[metabase.lib.schema.ref]].
+(lib.hierarchy/derive :metric ::aggregation)
```

Comment:
Hardcoding the list of clauses that were allowed inside aggregations was icky... I tweaked the schema to use the hierarchy instead.